chat-plugins/poll.js

```diff
@@ -0,0 +1,163 @@
+		this.totalVotes++;
+
+		this.update();
+	};
+
+	Poll.prototype.generateVotes = function () {
+		var output = '<div class="infobox" style="text-align:center"><p style="font-weight:bold;font-size:14pt">' + Tools.escapeHTML(this.question) + '</p>';
+		this.options.forEach(function (option, number) {
+			output += '<button value="/poll vote ' + number + '" name="send">' + number + '. ' + Tools.escapeHTML(option.name) + '</button><br/>';
+		});
+		output += '</div>';
+
+		return output;
+	};
+
+	Poll.prototype.generateResults = function () {
+		var output = '<div class="infobox" style="text-align:center"><p style="font-weight:bold;font-size:14pt">' + Tools.escapeHTML(this.question) + '</p>';
+		var iter = this.options.entries();
+
+		var i = iter.next();
+		while (!i.done) {
+			output += '<span style="font-weight:bold;padding:2px">' + i.value[0] + '. ' + Tools.escapeHTML(i.value[1].name) + '</span><br/><span style="font-size:8pt;padding:0px">' + i.value[1].votes + ' votes (' + Math.round((i.value[1].votes * 100) / this.totalVotes) + '% of total).</span><br/>';
+			i = iter.next();
+		}
+		output += '</div>';
+
+		return output;
+	};
+
+	Poll.prototype.update = function () {
+		var results = this.generateResults();
+
+		// Update the poll results for everyone that has voted
+		for (var i in this.room.users) {
+			var user = this.room.users[i];
+			if (this.voters.has(user.latestIp)) {
+				user.sendTo(this.room, '|uhtmlchange|poll' + this.room.pollNumber +  '|' + results);
+			}
+		}
+	};
+
+	Poll.prototype.display = function (user, broadcast) {
+		var votes = this.generateVotes();
+		var results = this.generateResults();
+
+		var target = {};
+
+		if (broadcast) {
+			target = this.room.users;
+		} else {
+			target[0] = user;
+		}
+
+		for (var i in target) {
+			var thisUser = target[i];
+			if (this.voters.has(thisUser.latestIp)) {
+				thisUser.sendTo(this.room, '|uhtml|poll' + this.room.pollNumber +  '|' + results);
+			} else {
+				thisUser.sendTo(this.room, '|uhtml|poll' + this.room.pollNumber +  '|' + votes);
+			}
+		}
+	};
+
+	Poll.prototype.end = function () {
+		var results = this.generateResults();
+
+		this.room.send('|uhtml|poll' + this.room.pollNumber +  '|' + results);
+		this.room.send('|uhtmlchange|poll' + this.room.pollNumber +  '|' + results);
+	};
+
+	return Poll;
+})();
+
+exports.commands = {
+	poll: {
+		new: function (target, room, user) {
+			var params = target.split(target.includes('|') ? '|' : ',').map(function (param) { return param.trim(); });
+
+			if (!this.can(permission, null, room)) return false;
+			if (room.poll) return this.errorReply("There is already a poll in progress in this room.");
+
+			if (params.length < 3) {
+				return this.errorReply("Not enough arguments for /poll new.");
+			}
+
+			var options = [];
+
+			for (var i = 1; i < params.length; i++) {
+				options.push(params[i]);
+			}
+
+			if (options.length > 8) {
+				return this.errorReply("Too many options for poll (maximum is 8).");
```

Comment:
Is there a particular reason for this limit at 8?

Perhaps a command that a room owner could change the maximum options cap would be a good idea, where 8 is still the default limit?